### DISCUSSION TITLE
Pictures disappeared after reboot

### DISCUSSION THREAD

**Discussion by GaGaius** [requarks/wiki#5266] [Error / Bug Report]
Hello,

I have installed wiki.js with SQLServer on my win10 server.
Everything is working fine, but when I restart the server, all images created with "Insert Asset" are gone. Only the Broken Link icon is displayed.

In the meantime I updated to version 2.5.280, but that didn't help.

In the log it says for each access to an image:
**[MASTER] error: path must be absolute or specify root to res.sendfile** 

What am I doing wrong?

Thanks

Edit: moved to Error / Bug Report

**Comment by mardsouza** [comment]
I'm having the same problem. This is the third time I've noticed it.

**Comment by GaGaius** [comment]
I don't understand. 
There seem to be several with this problem, why can't anyone give a hint?

**Comment by jarrodlovas** [comment]
Is it the same files? or are you noticing that once you upload the files and reboot, that they are missing altogether after the reboot?

**Reply by GaGaius** [reply]
The server is automatically restarted on a weekly basis.
After that, all links point to nothing.

No link works anymore.
I don't know if the files are still there, but they are not referenced correctly.

Thanks for your contribution

**Reply by jarrodlovas** [reply]
What is your installation method? I've seen issues previously with our images are installed inside a container getting lost after each reboot

**Reply by GaGaius** [reply]
I installed Wiki.js as described in https://docs.requarks.io/install/windows. 
No Docker or similar.

**Reply by jarrodlovas** [reply]
What database are you using?

**Reply by GaGaius** [reply]
Microsoft SQL Server

**Reply by GaGaius** [reply]
Microsoft SQL Server 2016

**Comment by jecksonbr** [comment]
We got the same in our company. When container is restarted, images broke.

I've checked paths, permissions, file owner... everything is fine.

File paths aren't changing when I restart the container, it's confusing why they're broking.

Path example:

`/var/lib/docker/overlay2/1a1e0b336be67c72daf38f05f89a291994b7ad7fb1d79d0e687d6d6c141c9a3e/diff/wiki/data/repo/imagens:
total 100K
-rw-r--r-- 1 ubuntu ubuntu 24K May 25 13:29 jeckson-2022.jpg`

nginx access.log:
`54.86.99.139 - - [27/May/2022:17:36:39 +0000] "GET /imagens/jeckson-2022.jpg HTTP/1.1" 500 21 "https://wiki.aquare.la/pt-br/AQ-DEVSECOPS/Projeto" "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/100.0.4896.75 Safari/537.36"`



External images don't face this problem.

Server: EC2 - Ubuntu 20.04
Running on Docker version 20.10.16, build aa7e414

**Comment by kanebrands** [comment]
We are having the same issue after upgrading wiki.js to the latest version - I can't figure out why none of my images will display in articles and I get the same error on the console:

error: path must be absolute or specify root to res.sendFile

One error for each image on an article. Has anyone been able to resolve this? The images files are back in the root after the upgrade (upgrade has the folder deleted and completely replaced) but they do not show up and telling the browser to view the image directly loads the correct path however it only shows "internal server error" when trying to render the image.

**Comment by kanebrands** [comment]
I was able to fix this error by modifying the server/models/assets.js file as follows:

Change the line at 204 (in static async getAssetFromCache function) from:
`await sendFile(cachePath, { dot files: 'deny' })`

To:
`try {

  await sendFile(cachePath, { dot files: 'deny' })

}catch (error) {

  await sendFile(cachePath, { dot files: 'deny', root: '/var/www/wiki/' })

}`

Note that the /var/www/wiki/ needs to match the root folder where your wiki lives on your system. This was snagged from another discussion thread here: https://github.com/requarks/wiki/discussions/3322

I'm just adding some additional context to the answer provided within that thread - wiki.js should pull this code change in and make the path pulled from the system or a config.yml setting.

**Reply by JuanAlisson00** [reply]
Thanks bro...

It is working....

**Comment by GaGaius** [comment]
Thank you for the suggestion.

I had to replace "dot files" with "dotfiles" but then wiki.js started.
For testing I inserted a new asset, but after restarting the server again only a broken link was visible.

Did this solution maybe help with others?

**Reply by aknotek** [reply]
Not sure if this has been solved elsewhere, but I had the same problem as you, @GaGaius. I updated the getAssetFromCache() function to just always return false and my images showed up. Assuming that the files may be served slower without this cache function enabled, but for my use case there aren't enough requests to make a difference.

`static async getAssetFromCache(assetPath, cachePath, res) {
    return false
}
`

**Reply by turbina358** [reply]
Works perfectly in 2024. Just keep in mind that you may have to do that after each update.

**Comment by Emerson-B-Ribeiro** [comment]
Good evening, I just installed the wiki on an Ubuntu server, and every time I restart the server, the images remain as a broken link and do not open, can anyone help me?

**Comment by hmawla** [comment]
same issue here... And there don't seem to be a solution apparently!!!

**Comment by NGPixel** [comment]
Try to "Flush Pages and Assets Cache", from Admin > Utilities > Flush Cache.
